service/metadata.yaml

```diff
@@ -71,3 +71,89 @@ telemetry:
       gauge:
         async: true
         value_type: int
+
+    receiver.produced.items:
+      enabled: true
+      description: Number of items emitted from the receiver.
+      unit: "{item}"
+      sum:
+        value_type: int
+        monotonic: true
+    processor.consumed.items:
+      enabled: true
+      description: Number of items passed to the processor.
+      unit: "{item}"
+      sum:
+        value_type: int
+        monotonic: true
+    processor.produced.items:
+      enabled: true
+      description: Number of items emitted from the processor.
+      unit: "{item}"
+      sum:
+        value_type: int
+        monotonic: true
+    connector.consumed.items:
+      enabled: true
+      description: Number of items passed to the connector.
+      unit: "{item}"
+      sum:
+        value_type: int
+        monotonic: true
+    connector.produced.items:
+      enabled: true
+      description: Number of items emitted from the connector.
+      unit: "{item}"
+      sum:
+        value_type: int
+        monotonic: true
+    exporter.consumed.items:
```

Comment:
Does this also need a produced items?

Reply:
It might be nice to have that metric eventually but this PR and the RFC which it works towards did not attempt to define how we would measure data emitted by exporters or consumed by receivers. It focuses only on data passed in between components because this is where we have the opportunity to apply measurements in a normalized way.